syntax = "proto3";

package helloworld;

service Greeter {
    rpc SayHello(HelloRequest) returns (HelloReply) {}
}

message HelloRequest {
    string str_val = 1;
    optional string optional_str_val = 2;
}

message HelloReply {
    string message = 1;
}
